addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-mount-service.js

```diff
@@ -93,6 +93,30 @@ class EnvironmentMountService extends Service {
     return this._updateResourcePolicies({ updateAwsPrincipals, workspaceRoleArn, s3Prefixes });
   }
 
+  async updateStudyFileMountIPAllowList(requestContext, existingEnvironment, ipAllowListAction) {
+    const [storageGatewayService, studyService] = await this.service(['storageGatewayService', 'studyService']);
+    // Check if the mounted study is using StorageGateway
+    const studiesList = await studyService.listByIds(
+      requestContext,
+      existingEnvironment.studyIds.map(id => {
+        return { id };
+      }),
+    );
+
+    // If yes, get the file share ARNs and call to update IP allow list
+    const fileShareARNs = studiesList.map(study => study.resources[0].fileShareArn).filter(arn => !_.isUndefined(arn));
+    if (!_.isEmpty(fileShareARNs)) {
+      let ip;
+      // If IP is in ipAllowListAction, use that, if not, find it in existingEnvironment
+      if ('ip' in ipAllowListAction) {
```

Comment:
Is it deliberate that string 'ip' needs to be in ipAllowListAction

Reply:
When it's add new IP to the allow list, the calling method usually knows about the new IP that should be added, so it's passed in here to avoid the trouble of downstream method having the find out the new IP again; and when it's remove old IP, the IP should be the one that's in the old DDB record.